test/acquisition/multi_objective/test_monte_carlo.py

```diff
@@ -1624,3 +1627,28 @@ def get_acqf(model, matheron):
                     rtol=1e-4,
                 )
             )
+
+    def test_with_transformed(self):
+        # Verify that _set_sampler works with transformed posteriors.
+        mm = MockModel(
+            posterior=PosteriorList(
+                TransformedPosterior(
+                    MockPosterior(samples=torch.rand(2, 3, 1)), lambda X: X
+                ),
+                TransformedPosterior(
+                    MockPosterior(samples=torch.rand(2, 3, 1)), lambda X: X
+                ),
+            )
+        )
+        sampler = ListSampler(
+            IIDNormalSampler(sample_shape=torch.Size([2])),
+            IIDNormalSampler(sample_shape=torch.Size([2])),
+        )
+        # This calls _set_sampler which would previously fail.
```

Comment:
previously w.r.t to what? This PR?

Reply:
Yep

Reply:
Maybe let's update the comment with what the test does specifically rather than saying "previously"? The context will get lost.